crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -582,18 +609,20 @@ impl<'db> TypeInferenceBuilder<'db> {
         }
     }
 
+    #[tracing::instrument(skip_all)]
     fn infer_parameter_with_default(&mut self, parameter_with_default: &ast::ParameterWithDefault) {
         let ast::ParameterWithDefault {
             range: _,
             parameter,
             default: _,
         } = parameter_with_default;
 
-        self.infer_optional_expression(parameter.annotation.as_deref());
+        self.infer_optional_annotation_expression(parameter.annotation.as_deref());
```

Comment:
This is a good thing to do, I *believe*, but it should be in the *next* PR, which will wire up very basic parameter inference.

```suggestion
        self.infer_optional_expression(parameter.annotation.as_deref());
```